src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java

```diff
@@ -35,16 +36,15 @@ public class Resource2MarcRecordMapperImpl implements Resource2MarcRecordMapper
   private final MarcFactory marcFactory;
   private final Collection<Bibframe2MarcFieldRuleApplier> rules;
   private final List<Ld2MarcMapper> ld2MarcMappers;
-  private final DataFieldPostProcessor dataFieldPostProcessor;
   private final Comparator<Subfield> subfieldComparator;
+  private final Supplier<DataFieldPostProcessor> dataFieldPostProcessorSupplier;
 
   @Override
   public Record toMarcRecord(Resource resource) {
     var marcRecord = marcFactory.newRecord();
     var cfb = new ControlFieldsBuilder();
     var dataFields = getFields(new ResourceEdge(null, resource, null), cfb);
-    var combinedFields = dataFieldPostProcessor.apply(dataFields);
```

Comment:
Here, we were combining the MARC data fields after generating all the MARC fields. At this stage, we don't know which MARC fields should be combined and which should not. So, removed this logic from here